File: ReactNativeApp/ios/Modules/SSVC/Instance/Points/PointsBuilder.swift
//
//  PointsBuilder.swift
//  jogo
//
//  Created by Mohsin on 13/10/2021.
//

import Foundation

public class PointsBuilder: SSVCBuilder {
  
  private static func buildState() -> PointsState {
    return PointsState()
  }
  
  public static func buildSquared() ->
  SSVC<PointsState, SquaredPointsView, PointsController> {
      
    let state = buildState()
    let squaredPointsView = SquaredPointsView(state: state)
    let pointsController = PointsController(state: state)
    return SSVC(state: state, stateView: squaredPointsView, controller: pointsController)
  }
}

File: ReactNativeApp/android/app/src/main/java/com/components/basic/points/PointsBuilder.java
package com.components.basic.points;

import com.ssvc.SSVC;
import com.ssvc.SSVCBuilder;

public class PointsBuilder extends SSVCBuilder {

    static private PointsState buildState() {
        return new PointsState();
    }

    public static SSVC<PointsState, PointsView, PointsController> buildPoints() {
        PointsState state = buildState();
        PointsView pointsView = new PointsView(state);
        PointsController pointsController = new PointsController(state);
        return new SSVC<>(state, pointsView, pointsController);
    }

}
